content/posts/webpack/index.md

```diff
@@ -0,0 +1,103 @@
+---
+date: "2015-07-07"
+title: Webpack
+tags:
+  - javascript
+  - task-runner
+authors:
+  - MoOx
+header:
+  credit: http://webpack.github.io/
+  linearGradient: 160deg, rgb(204, 51, 51), rgba(204, 51, 51, .6)
+---
+
+# Pourquoi webpack ?
+
+Avant webpack, nous avions beau avoir des super task-runners comme
+[grunt](/posts/js/premiers-pas-avec-grunt)
+ou
+[gulp](/posts/js/introduction-gulp),
+il faut avouer qu'on bricolait quand même pas mal.
+
+Pourquoi ? Devoir ajuster des configurations/définitions de tâches pour ajuster
+les copies d'images ou de fonts… Ou encore ajuster les urls de générations…
+Il y avait avec ces solutions une grande partie de rafistolage, où l’on se
+devait de faire très attention à la moindre réorganisation de code, sous peine
+de casser une partie du rendu.
+
+Prenons par exemple une image de fond déclaré en CSS. Qui vous préviendra qu'une
+référence n'est pas bonne (à part peut-être vos logs de serveur web en
+recherchant les pages d'erreurs 404) ?
+
+Vous avez la responsabilité de gérer vos tâches et leurs résultats tout en vous
+assurant du bon fonctionnement car vous êtes la glu entre tous ces morceaux.
+
+# Webpack: faire un pack prêt pour le web
+
+Pas besoin de s'appeler Enstein pour comprendre les intentions de webpack en
+interprétant le nom: webpack.
+
+Nativement, webpack s'occupe uniquement de ressources JavaScript.
+Les *loaders* permettent de transformer tout et n'importe quoi en JavaScript
+(mais pas que). Ainsi, tout est consommable en tant que module.
+
+C'est cette partie qui est aujourd'hui la plus intéressante et la plus flexible.
+
+<figure>
+  ![](index.jpg)
+  <figcaption>
+    webpack transforme une multitude de fichier en lots par responsabilité
+  </figcaption>
+</figure>
+
+Webpack, bien que principalement orienté vers les modules JavaScript, va
+donc pouvoir s'occuper de toutes les ressources dont vous pourrez avoir besoin.
```

Comment:
va donc pouvoir s'occuper aussi bien de vos modules javascript, mais aussi de vos modules css ([yes, it's a thing!](https://github.com/css-modules/css-modules)), et aussi de vos fichiers svg, gif, png, etc. ainsi que potentiellement n'importe quel type de fichier pour peu que vous preniez le temps d'écrire un `loader` (mais rassurez vous, il existe déjà un loader pour tous les fichiers que vous manipulez).